### DISCUSSION TITLE
API that returns a list of Active LCD Endpoints

### DISCUSSION THREAD

**Discussion by TheArchitect123** [classic-terra/core#146] [Ideas]
**Problem definition**
Can we please have an endpoint (FCD) that returns a list of active LCD endpoints (base urls)? It avoids having to ask each time the devs in this org for it if the urls change, or a hosting provider closes it down.

**Comment by ZaradarBH** [comment]
Hey buddy. First of all I personally think its a great idea, but I am  not convinced that the FCD indexer is the correct application for this job. Primarily due to the nature of how it works. Its basically just a NodeJS application that stores data from blocks in a Postgres DB and if it has maintain an up-2-date index of the active LCD endpoints in the network it would introduce quiet a lot of unnecessary traffic as all the FCD nodes would effectively engage in "long polling" the validators to ensure that  they are active and alive. It would probably be better to extend the oracle module to enable validators to provide a active LCD link as part of their oracle-feeder voting process, which occurs every 5 blocks, and then simply extend the message server in the oracle module to expose a few new LCD operations to enable SDKs to query the  L1 for active LCDs so you can build in a client-side load-balancer as we discussed in the past.  As such I think it would not take much time to extend the oracle-feeder to add in the extra information, I can show you in a session if you want to get a better understanding of its internals and extending the Oracle keeper with an array to store the LCD meta-data is a few days of coding and testing. So I recon overall we could mock something up in about a week. However it might make more sense to  do this in Q2 as we are currently trying to stay focused on  the next upgrade. :)

**Comment by ZaradarBH** [comment]
Removed [FEATURE] prefix. We use labels instead, [FEATURE] == "enhancement" :)
